File: go/easy/shuffle.go
package main

func shuffle(nums []int, n int) []int {
	rs := make([]int, len(nums))
	mid := len(nums) / 2
	for i := 0; i < len(rs); i++ {
		if i%2 == 0 {
			rs[i] = nums[i/2]
		} else {
			rs[i] = nums[mid+i/2]
		}
	}
	return rs
}

File: rust/src/easy/shuffle.rs
pub fn shuffle(nums: Vec<i32>, n: i32) -> Vec<i32> {
    let mut ans = Vec::with_capacity(nums.len());
    for i in 0..n as usize {
        ans.push(nums[i]);
        ans.push(nums[i + n as usize]);
    }
    ans
}